pkgs/tools/misc/lorri/default.nix

```diff
@@ -27,19 +27,19 @@ rustPlatform.buildRustPackage rec {
     owner = "target";
     repo = pname;
     # Run `eval $(nix-build -A lorri.updater)` after updating the revision!
-    rev = "03f10395943449b1fc5026d3386ab8c94c520ee3";
-    sha256 = "0fcl79ndaziwd8d74mk1lsijz34p2inn64b4b4am3wsyk184brzq";
+    rev = "7b84837b9988d121dd72178e81afd440288106c5";
+    sha256 = "0rkga944jl6i0051vbsddfqbvzy12168cbg4ly2ng1rk0x97dbr8";
   };
 
-  cargoSha256 = "1daff4plh7hwclfp21hkx4fiflh9r80y2c7k2sd3zm4lmpy0jpfz";
+  cargoSha256 = "0k7l0zhk2vzf4nlwv4xr207irqib2dqjxfdjk1fprff84c4kblx8";
   doCheck = false;
 
   BUILD_REV_COUNT = src.revCount or 1;
   RUN_TIME_CLOSURE = pkgs.callPackage ./runtime.nix {};
 
   nativeBuildInputs = with pkgs; [ nix direnv which ];
```

Comment:
Also, this looks weird - why do we need `nix` and `direnv` as `nativeBuildInput`?

Reply:
`nix` and `direnv` are runtime dependencies. Is there a better way of declaring that dependency?

Reply:
`nativeBuildInputs` is useful to distinguish between platforms w.r.t cross-compiling. Do we execute `nix` or `direnv` during build? The build might also work without `which`.

Does `lorri` rely on `nix` and `direnv` being available during runtime? Or is it only `nix`? It might be sufficient to have it available in the systemd service path - in interactive shells `nix` should already be available in `$PATH`.

Reply:
`lorri` requires `nix` at runtime. Some integration tests require `direnv`, but we don't run them on Hydra at the moment. Similarly for `which`, though it looks like the integration tests only claim to use it but don't really: https://github.com/target/lorri/blob/7b84837b9988d121dd72178e81afd440288106c5/tests/integration/envrctestcase.rs#L19-L23

> It might be sufficient to have it available in the systemd service path - in interactive shells nix should already be available in `$PATH`.

So you're saying that there is no need to have `nix` in `nativeBuildInputs`? Cool, I'll remove it.